### DISCUSSION TITLE
Line weight

### DISCUSSION THREAD

**Discussion by Kelv7788** [sumatrapdfreader/sumatrapdf#3246] [Ideas]
Hello
There are very thin lines in PDF files converted from DWG drawings so it is difficult to read the text and lines.
Is it possible to add the line weight parameter to configure the best line thickness for any monitor?

I am sending the sample with bigger line thickness in another PDF viewer for comparison with Sumatra PDF viewer. 
![20230128_225735](https://user-images.githubusercontent.com/78591633/215288763-9b062a61-f7ef-45ea-a416-4d3857d6c215.png)

**Comment by GitHubRulesOK** [comment]
This has been covered several times that Artifex Ghostscript & MuPDF (the core of SumatraPDF) attempt not to make the lines thicker than their definition. As a Senior Draughtsman I had to ensure my staff knew the desired minimum thickness e.g. never describe lines as zero wide but increase in line with the internationally or regionally or industry recognise Standards (never thinner never thicker)

It is possible to correct draughtspersons errors in SOME cases but not all, if you want sub-standard disproportionate thickening  then Edge may help to thicken them during reprint. One was corrected here (removed for privacy) https://github.com/sumatrapdfreader/sumatrapdf/issues/1076 and another here https://github.com/sumatrapdfreader/sumatrapdf/issues/1145
![image](https://user-images.githubusercontent.com/12754913/215323922-c42825c5-21a4-4af4-a42e-a5b8c947e058.png)

**Edge**
![image](https://user-images.githubusercontent.com/12754913/215292110-d0745692-58e1-4ac8-9fab-5ea407a7fb09.png)
**SumatraPDF**
![image](https://user-images.githubusercontent.com/12754913/215292152-18527461-2f5e-4bb6-a043-800fc97427dc.png)



Some past file types were possibly not well handled but that was resolved in this thread. https://github.com/sumatrapdfreader/sumatrapdf/issues/1034

For different issues see thos linked such as https://github.com/sumatrapdfreader/sumatrapdf/discussions/2781

**Comment by Kelv7788** [comment]
Thank you for your clarification.
I have found many mentionings regarding the thin lines on another forums also.
This means that many users need to configure the line thickness.
It looks like the most correct way is to add variable to set the minimum line thickness.
The default value will allow to correspond to Standards (never thinner never thicker), and users will be able to make the lines thicker as they need.
This will allow to see the lines on new 4K an 8K monitors with pixel size 0.16 mm.
Windows have the scale feature for the same issue.

There are even better example in the thread [#1034](https://github.com/sumatrapdfreader/sumatrapdf/issues/1034)
I have seen the same issue many times. 
It is very important not to loose the details in different industries.
So it is better to see something then nothing.
